### PR TITLE
Demo

### PR DESCRIPTION
**Pull request by Ecksmas**
Finalized media functions

### CODE REVIEW

**Review by rymdolle — CHANGES_REQUESTED** [review]
Don't auto format code that is not changed otherwise

File: src/audio.cpp [hunk 1]
```diff
@@ -5,18 +5,22 @@
...
+                            20, "I2S BCK",
+                            21, "I2S LRCK"));
+
+Audio::Audio()
+{
   decoder_ = MP3InitDecoder();
 
   // Set sample rate and set mono or stereo audio
   audio_format_ = {
-    .sample_freq = 48000,
```
Comment (rymdolle) [inline src/audio.cpp]: Formatted without change

File: src/audio.cpp [hunk 2]
```diff
@@ -28,23 +32,26 @@ Audio::Audio() {
   buffer_pool_ = audio_new_producer_pool(&producer_format_, 10, kMaxFrameSize);
 }
 
-Audio::~Audio() {
```
Comment (rymdolle) [inline src/audio.cpp]: Formatted without change

File: src/audio.cpp [hunk 3]
```diff
@@ -56,17 +63,19 @@ bool Audio::init_i2s() {
   return false;
 }
 
-int Audio::stream_decode(uint8_t *data, int size) {
```
Comment (rymdolle) [inline src/audio.cpp]: Formatted without change

File: src/audio.cpp [hunk 4]
```diff
@@ -76,17 +85,41 @@ int Audio::stream_decode(uint8_t *data, int size) {
   MP3FrameInfo info;
   MP3GetLastFrameInfo(decoder_, &info);
 
-  if (info.nChans == 1) {
```
Comment (rymdolle) [inline src/audio.cpp:79]: Formatted without change

File: src/http_client.cpp [hunk 5]
```diff
@@ -4,80 +4,101 @@
...
+
+  // Function to handle data being transmitted
+  err_t http::client::recv(void *arg, struct altcp_pcb *pcb, struct pbuf *p, err_t err)
+  {
+    http::request *req = reinterpret_cast<http::request *>(arg);
     uint16_t offset = 0;
 
     if (p == NULL)
     {
-        printf("No data received or connection closed\n");
```
Comment (rymdolle) [inline src/http_client.cpp:43]: Formatted without change

**Review by rymdolle — APPROVED** [review]

**Review by Lapingold — COMMENTED** [review]
Lgtm

**Review by Lapingold — APPROVED** [review]

**Pull request merged**